File: server/computed_functions/02_provider_api_keys.compute.py
"""02_provider_api_keys — STARTUP. Composite map for {{fn:provider_api_keys.<provider>}}."""
from __future__ import annotations
import os


def _from_env(name: str) -> str:
    return os.environ.get(name, f"(unset:{name})")


def compute(_ctx, _path=None):
    return {
        "gemini_openai": _from_env("GEMINI_API_KEY"),
        "openai": _from_env("OPENAI_API_KEY"),
        "openai_embeddings": _from_env("OPENAI_API_KEY"),
        "anthropic": _from_env("ANTHROPIC_API_KEY"),
        "figma": _from_env("FIGMA_API_TOKEN"),
        "github": _from_env("GITHUB_API_TOKEN"),
        "jira": _from_env("JIRA_API_TOKEN"),
        "confluence": _from_env("CONFLUENCE_API_TOKEN"),
        "saucelabs": _from_env("SAUCE_ACCESS_KEY"),
        "servicenow": _from_env("SERVICENOW_PASSWORD"),
        "rally": _from_env("RALLY_API_KEY"),
        "statsig": _from_env("STATSIG_API_KEY"),
        "sonar": _from_env("SONAR_TOKEN"),
    }


scope = "STARTUP"

File: server/computed_functions/02_provider_api_keys.compute.mjs
// 02_provider_api_keys — STARTUP. Composite returning { gemini_openai, openai, anthropic, ... }.
// Accessed by server.dev.yaml as {{fn:provider_api_keys.<provider>}}. Each value is the
// corresponding *_API_KEY env var, or a "(unset:<name>)" placeholder so the resolver fills
// the field instead of leaving the literal template string.
const fromEnv = (name) => process.env[name] ?? `(unset:${name})`;

export default function compute(_ctx, _path) {
  return {
    gemini_openai: fromEnv("GEMINI_API_KEY"),
    openai: fromEnv("OPENAI_API_KEY"),
    openai_embeddings: fromEnv("OPENAI_API_KEY"),
    anthropic: fromEnv("ANTHROPIC_API_KEY"),
    figma: fromEnv("FIGMA_API_TOKEN"),
    github: fromEnv("GITHUB_API_TOKEN"),
    jira: fromEnv("JIRA_API_TOKEN"),
    confluence: fromEnv("CONFLUENCE_API_TOKEN"),
    saucelabs: fromEnv("SAUCE_ACCESS_KEY"),
    servicenow: fromEnv("SERVICENOW_PASSWORD"),
    rally: fromEnv("RALLY_API_KEY"),
    statsig: fromEnv("STATSIG_API_KEY"),
    sonar: fromEnv("SONAR_TOKEN"),
  };
}

export const scope = "STARTUP";
